### PR TITLE
run: check that the binary exists, error if not, warn if symlink

### PR DESCRIPTION
**Pull request by dgonyeo**
acbuild run will now check if the given binary exists, and will print a
nice error message if it doesn't, as opposed to relying on
systemd-nspawn to inform the user.

Additionally on systems where systemd's version is <228, systemd may
error out when attempting to exec a symlink. This adds a check for this
case, and prints a warning.

Fixes https://github.com/appc/acbuild/issues/165. Fixes https://github.com/appc/acbuild/issues/159.

New behavior when binary doesn't exist:
```
[nix-shell:~/go/src/github.com/appc/acbuild]$ sudo ./bin/acbuild run /bin/sh
/bin/sh: can't access tty; job control turned off
/ # 
[nix-shell:~/go/src/github.com/appc/acbuild]$ sudo ./bin/acbuild run /bin/shz
run: the binary "/bin/shz" doesn't exist

[nix-shell:~/go/src/github.com/appc/acbuild]$ sudo ./bin/acbuild run shz
run: shz not found in any of: [/usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin]

[nix-shell:~/go/src/github.com/appc/acbuild]$ sudo ./bin/acbuild run sh
/bin/sh: can't access tty; job control turned off
/ # 
```

New behavior when binary is symlink and systemd < 228:
```
[root@localhost acbuild]# ./examples/apache/build-apache.sh 
Beginning build with an empty ACI
Setting name of ACI to example.com/apache
Adding dependency "quay.io/coreos/alpine-sh"
Running: [apk update]
Downloading quay.io/coreos/alpine-sh: [========================] 2.65 MB/2.65 MB
fetch http://dl-4.alpinelinux.org/alpine/v3.2/main/x86_64/APKINDEX.tar.gz
v3.2.3-185-g702b096 [http://dl-4.alpinelinux.org/alpine/v3.2/main]
OK: 5295 distinct packages available
Running: [apk add apache2]
(1/6) Installing libuuid (2.26.2-r0)
(2/6) Installing apr (1.5.2-r0)
(3/6) Installing expat (2.1.0-r1)
(4/6) Installing apr-util (1.5.4-r0)
(5/6) Installing pcre (8.37-r1)
(6/6) Installing apache2 (2.4.16-r0)
Executing busybox-1.23.2-r0.trigger
OK: 11 MiB in 21 packages
Running: [/bin/sh -c echo 'ServerName localhost' >> /etc/apache2/httpd.conf]
Warning: "/bin/sh" is a symlink, which systemd version 222 might error on
Directory /home/derek/go/src/github.com/appc/acbuild/.acbuild/target lacks the binary to execute or doesn't look like a binary tree. Refusing.
run: exit status 1
Ending the build
```

### CODE REVIEW

**Comment by dgonyeo** [comment]
@jonboulle with these changes still want the Lstat moved into findCmdInPath?

File: lib/run.go [hunk 1]
```diff
@@ -149,6 +149,17 @@ func (a *ACBuild) Run(cmd []string, insecure bool) (err error) {
...
 	}
+
+	finfo, err := os.Lstat(path.Join(nspawnpath, abscmd))
+	switch {
+	case os.IsNotExist(err):
+		return fmt.Errorf("the binary %q doesn't exist", abscmd)
+	case err != nil:
+		return err
+	case finfo.Mode()&os.ModeSymlink != 0 && systemdVersion < 228:
+		fmt.Printf("Warning: %q is a symlink, which systemd version %d might error on\n", abscmd, systemdVersion)
```
Comment (jonboulle) [inline lib/run.go]: stderr?

Comment (jonboulle) [inline lib/run.go]: nit, systemd-nspawn version

**Comment by jonboulle** [comment]
Don't feel strongly about it. Guess it's a pretty edge case scenario. This seems fine.

**Pull request merged**